TASVideos/Pages/Profile/Settings.cshtml

```diff
@@ -148,45 +163,86 @@
 	<partial name="_ValidationScriptsPartial" />
 	<script>
 		let avatarBoxElem = document.getElementById('@Html.IdFor(m => m.Avatar)');
-		let previewElem = document.getElementById('avatar-preview');
 
 		document.addEventListener("DOMContentLoaded", validateAvatar);
-		avatarBoxElem.addEventListener('blur', toggleAvatarPreview);
-		document.getElementById('preview-btn').addEventListener('click', toggleAvatarPreview);
-
+		document.addEventListener("DOMContentLoaded", onGravatarToggle);
+		avatarBoxElem.addEventListener('input', generateAvatarPreview);
+		document.getElementById('gravatar-email').addEventListener('input', generateGravatarPreview)
 		let avatarImgElem = document.getElementById('avatar-img');
+		avatarImgElem.onload = validateAvatar;
+
+		Array.from(document.querySelectorAll('[name="UseGravatar"]')).forEach(elem => elem.addEventListener('click', onGravatarToggle));
 
-		function toggleAvatarPreview() {
+		function generateAvatarPreview() {
 			const avatar = document.getElementById('@Html.IdFor(m => m.Avatar)').value;
 			if (avatar) {
-				avatarImgElem.onload = validateAvatar;
 				avatarImgElem.src = avatar;
 			} else {
-				previewElem.classList.add('d-none');
+				avatarImgElem.src = '';
 				preventSave(false);
 			}
 		}
 
 		function validateAvatar() {
+			console.log('validating avatar requirements');
 			const maxWidth = @AvatarRequirements.Width;
 			const maxHeight = @AvatarRequirements.Height;
+			const descSection = document.getElementById('avatar-description');
 
 			const tooBig = avatarImgElem.width > maxWidth || avatarImgElem.height > maxHeight;
 			if (tooBig) {
-				previewElem.classList.add('alert');
-				previewElem.classList.add('alert-danger');
+				descSection.classList.add('alert');
+				descSection.classList.add('alert-danger');
```

Comment:
Setting an `alert` on a `<small>` element breaks the layout.
![image](https://github.com/TASVideos/tasvideos/assets/22375320/843791b0-c38e-4369-add8-08696cc3b0a9)